### PR TITLE
adding new scopes for alerts

### PR DESCRIPTION
**Pull request by olrakovs-zz**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json [hunk 1]
```diff
@@ -2114,6 +2114,130 @@
         }
       }
     },
+    "/providers/Microsoft.CostManagement/externalBillingAccounts/{externalBillingAccountId}/alerts": {
+      "get": {
+        "tags": [
+          "Alerts"
+        ],
+        "operationId": "Alerts_ListByexternalBillingAccounts",
```
Comment (nschonni) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]:
```suggestion
        "operationId": "Alerts_ListByExternalBillingAccounts",
```

**Comment by salameer** [comment]
requesting ARM review

Reassigning since Joel is OOF

**Comment by kpajdzik** [comment]
If you haven't taken a look at our new onboarding experience at [OpenAPIHub](http://aka.ms/openapihub), it's a convenient way to create your PR when you're copying from an existing API version or when you're editing your existing specs. If you have any feedback or questions, feel free to use the feedback button on top of the site for help. Thanks!

**Comment by kpajdzik** [comment]
@olrakovs, do you need any SDK other than Go? If so, we need to add missing readme files.

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json [hunk 2]
```diff
@@ -2114,6 +2114,130 @@
...
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/externalBillingAccountIdParameter"
+          },
+          {
+            "name": "$filter",
+            "description": "May be used to filter alerts by properties/definition/type, properties/definition/category, properties/definition/criteria, properties/costEntityId, properties/creationTime, properties/closeTime, properties/status, properties/source. Supported operators are 'eq','lt', 'gt', 'le', 'ge'.",
+            "in": "query",
+            "required": false,
```
Comment (kpajdzik) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: I think this is not needed (same for other parameters).

Comment (olrakovs) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: Why not? we did it like other scopes. We want to be consistent to other scopes

**Comment by olrakovs** [comment]
No , we I don’t

**Comment by jhendrixMSFT** [comment]
@olrakovs can you please add examples for these new APIs?  Also are these APIs live?

**Comment by olrakovs** [comment]
I've added examples, those api's in a preview mode

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json (hunk 2 above)
Comment (kpajdzik) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: Sure, no harm done, it's just redundant.

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json [hunk 3]
```diff
@@ -2672,6 +2880,102 @@
...
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/externalSubscriptionIdParameter"
+          },
+		  {
```
Comment (kpajdzik) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: Formatting

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json [hunk 4]
```diff
@@ -2374,6 +2498,90 @@
...
+    },
+    "/providers/Microsoft.CostManagement/externalSubscriptions/{externalSubscriptionId}/alerts/{alertId}": {
+      "get": {
+        "tags": [
+          "Alerts"
+        ],
+        "operationId": "Alerts_GetAlertByExternalSubscriptions",
+        "description": "Gets an alert for external subscription by alert ID.",
+        "externalDocs": {
+          "url": "https://docs.microsoft.com/en-us/rest/api/costmanagement/"
```
Comment (kpajdzik) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: This leads to 404 so far, I guess this is going to be updated once the service goes GA?

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json [hunk 5]
```diff
@@ -2672,6 +2880,102 @@
...
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/externalBillingAccountIdParameter"
+          },
+		  {
```
Comment (kpajdzik) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: Formatting

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/examples/ExternalBillingAccountAlert.json [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+            "source": "User",
+            "details": {
+                "amountDue": "500.0",
+                "billedAmount": "680.36"
+            },
+            "costEntityId": "123",
+            "creationTime": "2018-09-03T11:21:43.3495582+03:00",
+            "status": "Active",
+            "closeTime": "2018-09-10T11:21:43.3495582+03:00",
+			"modificationTime": "2018-12-05T10:04:36.0321998Z",
```
Comment (kpajdzik) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/examples/ExternalBillingAccountAlert.json]: Formatting

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/examples/ExternalBillingAccountAlertList.json [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+                    "source": "User",
+                    "details": {
+                        "amountDue": "500.0",
+                        "billedAmount": "680.36"
+                    },
+                    "costEntityId": "123",
+                    "creationTime": "2018-09-03T11:21:43.3495582+03:00",
+                    "status": "Resolved",
+                    "closeTime": "2018-09-10T11:21:43.3495582+03:00",
+					"modificationTime": "2018-12-05T10:04:36.0321998Z",
```
Comment (kpajdzik) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/examples/ExternalBillingAccountAlertList.json]: Formatting

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/examples/ExternalBillingAccountAlertList.json [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+                    "source": "Preset",
+                    "details": {
+                        "amountDue": "500.0",
+                        "billedAmount": "680.36"
+                    },
+                    "costEntityId": "abc",
+                    "creationTime": "2018-09-03T11:21:43.3495582+03:00",
+                    "status": "Active",
+                    "closeTime": "1970-01-01T00:00:00.0000000+00:00",
+					"modificationTime": "2018-12-05T10:04:36.0321998Z",
```
Comment (kpajdzik) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/examples/ExternalBillingAccountAlertList.json]: Formatting

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/examples/ExternalSubscriptionAlert.json [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+			"modificationTime": "2018-12-05T10:04:36.0321998Z",
+			"statusModificationUserName": "Ido Ben",
+			"statusModificationTime": "2018-12-05T10:04:36.0321998Z"
+          }
+        }
+      }   
+    }
+  }
+  
+
```
Comment (kpajdzik) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/examples/ExternalSubscriptionAlert.json]: Formatting

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/examples/ExternalSubscriptionAlert.json [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+            "source": "User",
+            "details": {
+                "amountDue": "500.0",
+                "billedAmount": "680.36"
+            },
+            "costEntityId": "123",
+            "creationTime": "2018-09-03T11:21:43.3495582+03:00",
+            "status": "Active",
+            "closeTime": "2018-09-10T11:21:43.3495582+03:00",
+			"modificationTime": "2018-12-05T10:04:36.0321998Z",
```
Comment (kpajdzik) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/examples/ExternalSubscriptionAlert.json]: Formatting

1.

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/examples/ExternalSubscriptionAlertList.json [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+                    "source": "User",
+                    "details": {
+                        "amountDue": "500.0",
+                        "billedAmount": "680.36"
+                    },
+                    "costEntityId": "123",
+                    "creationTime": "2018-09-03T11:21:43.3495582+03:00",
+                    "status": "Resolved",
+                    "closeTime": "2018-09-10T11:21:43.3495582+03:00",
+					"modificationTime": "2018-12-05T10:04:36.0321998Z",
```
Comment (kpajdzik) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/examples/ExternalSubscriptionAlertList.json]: Formatting

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/examples/ExternalSubscriptionAlertList.json [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+                    "source": "Preset",
+                    "details": {
+                        "amountDue": "500.0",
+                        "billedAmount": "680.36"
+                    },
+                    "costEntityId": "abc",
+                    "creationTime": "2018-09-03T11:21:43.3495582+03:00",
+                    "status": "Active",
+                    "closeTime": "1970-01-01T00:00:00.0000000+00:00",
+					"modificationTime": "2018-12-05T10:04:36.0321998Z",
```
Comment (kpajdzik) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/examples/ExternalSubscriptionAlertList.json]: Formatting

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json (hunk 4 above)
Comment (olrakovs) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: you are right

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/examples/ExternalSubscriptionAlertList.json (hunk 12 above)
Comment (olrakovs) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/examples/ExternalSubscriptionAlertList.json]: fixed

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/examples/ExternalSubscriptionAlertList.json (hunk 11 above)
Comment (olrakovs) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/examples/ExternalSubscriptionAlertList.json]: fixed

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/examples/ExternalSubscriptionAlert.json (hunk 10 above)
Comment (olrakovs) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/examples/ExternalSubscriptionAlert.json]: fixed

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/examples/ExternalSubscriptionAlert.json (hunk 9 above)
Comment (olrakovs) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/examples/ExternalSubscriptionAlert.json]: fixed

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/examples/ExternalBillingAccountAlertList.json (hunk 8 above)
Comment (olrakovs) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/examples/ExternalBillingAccountAlertList.json]: fixed

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/examples/ExternalBillingAccountAlertList.json (hunk 7 above)
Comment (olrakovs) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/examples/ExternalBillingAccountAlertList.json]: fixed

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/examples/ExternalBillingAccountAlert.json (hunk 6 above)
Comment (olrakovs) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/examples/ExternalBillingAccountAlert.json]: fixed

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json (hunk 5 above)
Comment (olrakovs) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2018-08-01-preview/costmanagement.json]: fixed

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]
There are model validation failures that must be fixed, see https://travis-ci.org/Azure/azure-rest-api-specs/jobs/517442323

**Review by KrisBash — APPROVED** [review]
LGTM

**Comment by jhendrixMSFT** [comment]
@AutorestCI regenerate azure-sdk-for-go

**Comment by jhendrixMSFT** [comment]
@olrakovs any update here?

**Comment by kpajdzik** [comment]
@olrakovs any updates?

**Comment by olrakovs** [comment]
Hi, due to Israel holidays, we won’t work until 27/4

Thanks
Oleg

**Comment by kpajdzik** [comment]
@olrakovs, let us know if this PR is abandoned

**Comment by olrakovs** [comment]
Hi, 
the PR is still in progress

**Comment by jhendrixMSFT** [comment]
@olrakovs any update?

**Comment by olrakovs** [comment]
It is still in progress
Thanks
Oleg

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by kpajdzik** [comment]
Please reopen when you're ready to merge.

**Pull request closed without merging**